components/articles/single-article.jsx

```diff
@@ -17,22 +19,56 @@ const SingleArticle = () => {
     });
   }, []);
 
+  function likeArticleButton(article_id) {
+    setLikeArticle(true)
+    increaseArticleVote(article_id);
+
+    setSingleArticle((currentArticle) => {
+      if (currentArticle.article_id === article_id) {
+        return { ...currentArticle, votes: currentArticle.votes + 1 };
+      }
+    });
+  }
+  function dislikeArticleButton (article_id) {
+    setLikeArticle(true)
+    decreaseArticleVote(article_id)
+
+    setSingleArticle((currentArticle) =>{
+        if (currentArticle.article_id === article_id){
+            return {...currentArticle, votes:currentArticle.votes-1}
+        }
+    })
+  }
+
```

Comment:
Hi Jenny, it's a bit difficult to tell the difference between your like and dislike button 👓 , is there a way you could make it a bit more visible - you could use something like react-icons?
Ideally you would be able to undo a like or dislike, but if this is the intended behaviour you wanted you don't need to change it.
Also you should add error handling to manage the situation where you lose internet by returning the votes to the correct value if the internet drops and the vote api call doesn't go through
https://notes.northcoders.com/courses/js-front-end/ui-behaviours-and-optimistic-rendering#error-handling